scripts/lintcommit.lua

```diff
@@ -86,12 +86,33 @@ local function validate_commit(commit_message)
       vim.inspect(allowed_types))
   end
 
-  -- Check if scope is empty
+  -- Check if scope is appropriate
   if before_colon:match("%(") then
     local scope = vim.trim(before_colon:match("%((.*)%)"))
+
     if scope == '' then
       return [[Scope can't be empty.]]
     end
+
+    if scope == "ts" or scope == "tree-sitter" then
+      return [[Scope should be "treesitter"]]
+    end
+
+    if scope == "hl" then
+      return [[Scope should be "highlight"]]
+    end
+
+    if vim.startswith(scope, "nvim_") then
+        return [[Scope should be "api"]]
+    end
+
+    if scope == "doc" or scope == "docs" then
+        return [[Prefer to use "docs" as a type rather than a scope. Example: docs(treesitter):...]]
+    end
+
+    if scope == "filetype.vim" or scope == "filetype.lua" then
```

Comment:
instead of lots of if-else, could define a "misspelling" table like this:

```
local misspelling = {
  ['filetype.vim'] = 'filetype',
  ['filetype.lua'] = 'filetype',
  ['doc'] = 'docs',
  ['tree-sitter'] = 'treesitter',
  ['ts'] = 'treesitter',
}

if misspelling[scope] then
    return ('scope should be "%s" instead of "%s"'):format(misspelling[scope], scope)
end

Reply:
`docs` is a type, not a scope.

Reply:
We are thinking it can be both. E.g. improvements to the HTML generator might be `feat(docs)`, whereas `docs` topic is for actual documentation changes.